### ISSUE TITLE
 Bash Brace Expansion

### ISSUE THREAD

**Issue by pando85** [xonsh/xonsh#2302]
I can't live without [brace expansion](http://www.linuxjournal.com/content/bash-brace-expansion).

**Comment by TeamSpen210** [comment]
You could define a function or two and stick it in builtins in your `xonshrc`, then invoke with `@(func())`. Macros would help there if you want to get rid of quotes around strings.

**Comment by pando85** [comment]
No, I'm talking about expand commands like this:
```bash
rm /a/long/path/{foo,bar} 
# expand as this:
rm /a/long/path/foo /a/long/path/bar
```

**Comment by melund** [comment]
There is an open (never finished) PR for this. https://github.com/xonsh/xonsh/pull/1379

Any help moving it along would be great.

**Comment by gforsyth** [comment]
This should probably go in the bashisms xontrib or a similar space.

**Comment by JohnLunzer** [comment]
I agree with @gforsyth I think bashisms sounds like a good home. It seems like enabling this feature would supplant python set notation syntax. 

That said, perhaps there isn't a reason to maintain the `{}` syntax.

Additionally, is there are more  "xonshic" way to accomplish this in similarly terse syntax?

**Comment by adqm** [comment]
In my fork, I made brace expansion happen inside of ``g`...` `` "strings" but not outside, making use of the [braceexpand](https://pypi.python.org/pypi/braceexpand) module for the actual implementation.  This avoids the conflict with dictionaries at the cost of the three extra characters (``g, `, and ` ``).

If that's a happy medium, I'm happy to port it over.

**Comment by JohnLunzer** [comment]
@adqm It sounds better than not having brace expansion at all. But I'd let others chime in on other possible solutions first.

**Comment by amstan** [comment]
> I can't live without [brace expansion](http://www.linuxjournal.com/content/bash-brace-expansion).

Indeed, one the frequent patterns I do is stuff like `mv /etc/pacman.conf{.pacnew,}`. I essentially type in the common filename, then the before,after inside the {}.
